src/gsy_e/models/strategy/future/strategy.py

```diff
@@ -0,0 +1,258 @@
+        initial_buying_rate: float, final_buying_rate: float, asset_type: AssetType
+) -> Union[TemplateStrategyUpdaterInterface, FutureTemplateStrategyBidUpdater]:
+
+    _update_interval = FutureTemplateStrategiesConstants.UPDATE_INTERVAL_MIN
+    if asset_type in [AssetType.CONSUMER, AssetType.PROSUMER]:
+        return FutureTemplateStrategyBidUpdater(
+            initial_rate=initial_buying_rate,
+            final_rate=final_buying_rate,
+            fit_to_limit=True,
+            energy_rate_change_per_update=None,
+            update_interval=duration(minutes=_update_interval),
+            rate_limit_object=min)
+    return TemplateStrategyUpdaterInterface()
+
+
+def future_strategy_offer_updater_factory(
+        initial_selling_rate: float, final_selling_rate: float, asset_type: AssetType
+) -> Union[TemplateStrategyUpdaterInterface, FutureTemplateStrategyOfferUpdater]:
+
+    _update_interval = FutureTemplateStrategiesConstants.UPDATE_INTERVAL_MIN
+    if asset_type in [AssetType.PRODUCER, AssetType.PROSUMER]:
+        return FutureTemplateStrategyOfferUpdater(
+            initial_rate=initial_selling_rate,
+            final_rate=final_selling_rate,
+            fit_to_limit=True,
+            energy_rate_change_per_update=None,
+            update_interval=duration(minutes=_update_interval),
+            rate_limit_object=max)
+    return TemplateStrategyUpdaterInterface()
+
+
+class FutureMarketStrategy(FutureMarketStrategyInterface):
+    """Manages bid/offer trading strategy for the future markets, for a single asset."""
+    def __init__(self, asset_type: AssetType,
+                 initial_buying_rate: float, final_buying_rate: float,
+                 initial_selling_rate: float, final_selling_rate: float):
+        """
+        Args:
+            initial_buying_rate: Initial rate of the future bids
+            final_buying_rate: Final rate of the future bids
+            initial_selling_rate: Initial rate of the future offers
+            final_selling_rate: Final rate of the future offers
+        """
+        super().__init__()
+        self._offer_updater = future_strategy_offer_updater_factory(
+            initial_selling_rate, final_selling_rate, asset_type
+        )
+
+        self._bid_updater = future_strategy_bid_updater_factory(
+            initial_buying_rate, final_buying_rate, asset_type
+        )
+
+    def event_market_cycle(self, strategy: "BidEnabledStrategy") -> None:
+        """
+        Should be called by the event_market_cycle of the asset strategy class, posts
+        settlement bids and offers on markets that do not have posted bids and offers yet
+        Args:
+            strategy: Strategy object of the asset
+
+        Returns: None
+
+        """
+        if not strategy.area.future_markets:
+            return
+        self._bid_updater.update_and_populate_price_settings(strategy.area)
+        self._offer_updater.update_and_populate_price_settings(strategy.area)
```

Comment:
What if the `_bid_updater`, `_offer_updater` are instances of `TemplateStrategyUpdaterInterface`?  This `update_and_populate_price_settings` doesn't exist there AFAIS

Reply:
This is exactly the point of constructing the TemplateStrategyUpdaterInterface . Essentially the factory method is used in order to disable the implementation of this class for some asset types, therefore to still call the methods but do nothing at all. For asset types that need these, the factory method will deal with creating not the interface but the actual updater class.

Reply:
But I don't see where we're defining the `update_and_populate_price_settings` for example in the `TemplateStrategyUpdaterInterface`, so I would not predict that it will do nothing, I would predict that it will break for calling up a method that doesn't exist
Thanks for bearing with me, I could have missed it

Reply:
There is probably something wrong with the diff for you too like with Hannes before. It is the first method of the TemplateStrategyUpdaterInterface https://github.com/gridsingularity/gsy-e/pull/1277/files/05a66ff4e693d529f6dcb0b9d936ad1929cc6412#diff-b9af14d73fabe24d5f22776a3741af78f0a06e27159aa95519bd6e82304cc336R38

Reply:
Weird indeed, I can now see it but for some reason these methods were not shown yesterday, good then